pkg/proxy/access/clusteraccess.go

```diff
@@ -0,0 +1,40 @@
+package access
+
+import (
+	"net/url"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ClusterAccess holds an information needed to access the namespace in a member cluster for the specific user
+type ClusterAccess struct { // nolint:revive
+	// APIURL is the Cluster API Endpoint for the namespace
+	apiURL url.URL
+	// SAToken is a token of the Service Account which represents the user in the namespace
```

Comment:
It's not a SA token representing the user anymore. It's a member client SA now.
Maybe let's rename it from `saToken` to `impersonatorToken`?